zyte_common_items/pipelines.py

```diff
@@ -53,3 +55,67 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropLowProbabilityItemPipeline:
+    """This pipeline drops an item if its probability, defined in the settings,
+    is less than the specified threshold.
+
+    There are two ways that thresholds can be defined per item type using
+    the `ITEM_PROBABILITY_THRESHOLDS` setting:
+
+    With paths to Item classes:
+        ITEM_PROBABILITY_THRESHOLDS = {
+            "zyte_common_items.items.Article": 0.1,
+            "zyte_common_items.items.Product": 0.2
+        }
+
+    With Item classes:
+        from zyte_common_items import Article, Product
+        ITEM_PROBABILITY_THRESHOLDS = {
+            Article: 0.1,
+            Product: 0.2
+        }
+    """
+
+    def __init__(self, crawler):
+        self.stats = crawler.stats
+        self.thresholds = {}
+        self.init_thresholds(crawler.spider)
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
+
+    def init_thresholds(self, spider):
+        from scrapy.utils.misc import load_object
+
+        thresholds_settings = spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {})
+        for item, threshold in thresholds_settings.items():
+            item_type = load_object(item) if isinstance(item, str) else item
+            self.thresholds[item_type] = threshold
+
+    def get_threshold(self, item, spider):
```

Comment:
```suggestion
    def get_threshold_for_item(self, item, spider):
```
minor: when reading the tests, I was expecting it to return the value set by DEFAULT_ITEM_PROBABILITY_THRESHOLD. Perhaps this can make it a bit clearer.